jobrunner/manage_jobs.py

```diff
@@ -125,7 +125,7 @@ def create_and_populate_volume(job):
     # `docker cp` can't create parent directories for us so we make sure all
     # these directories get created when we copy in the code
     extra_dirs = set(Path(filename).parent for filename in input_files.keys())
-    if config.LOCAL_RUN_MODE:
+    if config.LOCAL_RUN_MODE and not job.is_action_reusable:
```

Comment:
Is `job.commit` `"none"` and not `None` because of how values are written to SQLite? Or because `JobRequest.commit` should be `str` rather than, say, `Optional[str]`? Or something else?

Reply:
I'm wondering whether we can do without magic strings.

Reply:
Ah, now I come to look at it I think the only reason I ended up using `"none"` was because of this line here:
https://github.com/opensafely-core/job-runner/blob/64b037a96f4bf19914e6a78d4c6815bac2c1945e/jobrunner/create_or_update_jobs.py#L94

But I think we can just change that to:
```py
if not config.LOCAL_RUN_MODE and not job_request.commit
```

I think both `None` and `""` would work here as the database column is NULLable.

Agree that it would be better to do without magic strings if we can.

Reply:
I'm going to push two commits. One replaces `"none"` with `None`. One removes `Job.is_action_reusable` and instead tests the value of `Job.commit`. I think both resolve this issue.

Reply:
:+1: I'd be inclined to just test for falseyness though, rather than explicitly checking `is None`. I'm not 100% sure the None will round-trip successfully through the db and I don't think it gains us anything to not handle the empty string as well. If you're confident it works though then happy to leave as is.

Reply:
I'm going to push a commit that tests for falseyness, as I'm also not sure what the DB does.